src/iden3comm/handlers/common.ts

```diff
@@ -98,42 +98,68 @@ export const processZeroKnowledgeProofRequests = async (
   const zkpResponses = [];
 
   for (const proofReq of requestScope) {
-    if (!opts.supportedCircuits.includes(proofReq.circuitId as CircuitId)) {
-      throw new Error(`Circuit ${proofReq.circuitId} is not allowed`);
-    }
-
-    const query = proofReq.query;
-    const groupId = query.groupId as number | undefined;
-    const combinedQueryData = combinedQueries.get(groupId as number);
-    if (groupId) {
-      if (!combinedQueryData) {
-        throw new Error(`Invalid group id ${query.groupId}`);
+    let zkpRes: ZeroKnowledgeProofResponse;
+    try {
+      if (!opts.supportedCircuits.includes(proofReq.circuitId as CircuitId) && !proofReq.optional) {
```

Comment:
what is the reason to add  && !proofReq.optional to this check?
Does it make sense to move further in case you can't generate proof because of unsupported circuit?